cachito/workers/scm.py

```diff
@@ -133,14 +133,21 @@ def fetch_source(self):
         # The previous archive does not mean the one just before the request that
         # schedules current task. The only reason for finding out such a file is
         # to access the git history. So, anyone is ok.
-        previous_archive = max(
-            self.sources_dir.package_dir.glob("*.tar.gz"), key=os.path.getctime, default=None
+        previous_archives = sorted(
+            self.sources_dir.package_dir.glob("*.tar.gz"), key=os.path.getctime, reverse=True
         )
-
-        if previous_archive:
-            self.update_and_archive(previous_archive)
-        else:
-            self.clone_and_archive()
+        for previous_archive in previous_archives:
+            try:
+                self.update_and_archive(previous_archive)
+                return
+            except (
+                git.exc.InvalidGitRepositoryError,
+                tarfile.ExtractError,
+                EnvironmentError,
```

Comment:
I was confused  about `EnvironmentError` for a second, turns out on python3.3 and later it is just an alias of `OSError`: https://docs.python.org/3.8/library/exceptions.html#EnvironmentError

Should we use `OSError` here instead? Would it also be possible to add a comment explaining when an `OSError` might occur?

Reply:
I checked the wrong reference when looking for exceptions to capture here. I checked https://github.com/python/cpython/blob/2.7/Lib/tarfile.py when I wanted to check https://github.com/python/cpython/blob/3.7/Lib/tarfile.py, where the EnvironmentError occurrences were substituted for ether OSError or ExtractError ones.

I will change it to `OSError`